### ISSUE TITLE
[BUG][SDW]Headset aplay with suspend & resume stress test output "Program params failed: -61"  error.

### ISSUE THREAD

**Issue by sinahuang** [thesofproject/linux#1344]
**Describe the bug**
Headset aplay with suspend & resume stress test output "Program params failed: -61"  error.

**To Reproduce:**
run command: aplay -Dhw:0,0 -r48000 -c2 -fs16_le /dev/zero & sudo ./stress-s3.sh 50 1 1

**frequency**
first try reproduce at 15th.
second try reproduce at first time.
third try reproduce at 8th.

**Expected result:**
No error should output during headset suspend & resume stress test .

**Actual result:**
Headset aplay with suspend & resume stress test output "Program params failed: -61"  error.
```
[  688.078012] acpi LNXPOWER:09: Turning OFF
[  688.078879] acpi LNXPOWER:08: Turning OFF
[  688.081071] OOM killer enabled.
[  688.081071] Restarting tasks ... 
[  688.082362]  SDW1-speakers: ASoC: prepare FE SDW1-speakers
[  688.082364]  SDW1-Playback: ASoC: prepare BE SDW1-Playback
[  688.082367] intel-sdw sdw-master-1: intel_prepare: SDW1 Pin2: start
[  688.082376] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[  688.085134] done.
[  688.085556] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[  688.085557] sdw_prepare_stream: SDW1 Pin2-Playback start
[  688.085650] intel-sdw sdw-master-1: Msg Ack not received
[  688.085651] intel-sdw sdw-master-1: Msg ignored for Slave 1
[  688.085653] rt700 sdw:1:25d:700:0:0: DPN_PortCtrl register write failed for port 1
[  688.085656] intel-sdw sdw-master-1: Program transport params failed: -61
[  688.085657] intel-sdw sdw-master-1: Program params failed: -61
[  688.085658] Prepare for stream:SDW1 Pin2-Playback failed: -61
[  688.085659] sdw_prepare_stream: SDW1 Pin2-Playback: done
[  688.085660] intel-sdw sdw-master-1: intel_prepare: SDW1 Pin2: done
[  688.085661] intel-sdw sdw-master-1: ASoC: cpu DAI prepare error: -61
[  688.085662]  SDW1-Playback: ASoC: backend prepare failed -61
[  688.085763]  SDW1-speakers: ASoC: hw_free FE SDW1-speakers
[  688.085767] sof-audio-pci 0000:00:1f.3: pcm: free stream 0 dir 0
[  688.085773]  SDW1-Playback: ASoC: hw_free BE SDW1-Playback
[  688.085777] intel-sdw sdw-master-1: intel_hw_free: SDW1 Pin2: start
[  688.085778] sdw_deprepare_stream: SDW1 Pin2-Playback: start
[  688.086172] sdw_deprepare_stream: SDW1 Pin2-Playback: done
[  688.086181] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[  688.086280] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[  688.086282] intel-sdw sdw-master-1: intel_hw_free: SDW1 Pin2: done
[  688.086306]  SDW1-Playback: ASoC: close BE SDW1-Playback
[  688.086308] intel-sdw sdw-master-1: intel_shutdown: SDW1 Pin2: start
[  688.086310] intel-sdw sdw-master-1: intel_shutdown: SDW1 Pin2: done
[  688.086337]  SDW1-speakers: ASoC: close FE SDW1-speakers
[  688.086341] sof-audio-pci 0000:00:1f.3: pcm: close stream 0 dir 0
[  688.086344]  SDW1-Playback: ASoC: BE SDW1-Playback event 2 dir 0
[  688.087094] intel-sdw sdw-master-1: Msg Ack not received
[  688.087096] intel-sdw sdw-master-1: Msg ignored for Slave 1
[  688.087300]  SDW1-speakers: ASoC: BE playback disconnect check for SDW1-Playback
[  688.087303]  SDW1-speakers: freed DSP playback path SDW1-speakers -> SDW1-Playback
[  688.130566] PM: suspend exit
[  689.025612] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[  689.025683] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
```
[s3_test_8_logger_all.log](https://github.com/thesofproject/linux/files/3749961/s3_test_8_logger_all.log)
[s3_test_8_logger_error.log](https://github.com/thesofproject/linux/files/3749962/s3_test_8_logger_error.log)
[test_8.log](https://github.com/thesofproject/linux/files/3749964/test_8.log)

**Environment**
Firmware: 3d5d314  https://github.com/thesofproject/sof/commits/master
kernel: 29debf8 https://github.com/thesofproject/linux/tree/integration/soundwire-latest
tplg: sof-cml-rt700.tglg
platform: CML RVP with codec ALC700 in SDW mode

**Comment by sinahuang** [comment]
this is split from bug #1311

**Comment by YvonneYang2** [comment]
Issue can also reproduce on ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode.

**Comment by YvonneYang2** [comment]
Issue reproduced on CML RVP with codec ALC700 in SDW mode.

Failed at 38th time.

Logs:
[s3_test_38_logger_all.log](https://github.com/thesofproject/linux/files/3765363/s3_test_38_logger_all.log)
[test_38.log](https://github.com/thesofproject/linux/files/3765364/test_38.log)

Test recipe:
kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 7c85d69
FW: https://github.com/thesofproject/sof/commits/master  commit: 8b059f6
tplg: sof-cml-rt700.tplg

**Comment by dengyangchao** [comment]
issue also reproduced on TGL RVP with codec ACL711, ALC1308 in SDW mode

Failed at 6th time

log:
[s3_test_6_logger_all.log](https://github.com/thesofproject/linux/files/3765432/s3_test_6_logger_all.log)
[test_6.log](https://github.com/thesofproject/linux/files/3765433/test_6.log)

Test recipe:
kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 15f27d8
FW: https://github.com/thesofproject/sof/commits/master commit: f277189
tplg: Same with TGL FW branch commit:  sof-tgl-rt711-rt1308-nohdmi.tplg rename to sof-tgl-rt711- 
         rt1308.tplg

**Comment by plbossart** [comment]
Seems like a problem with the codec trying to write registers before we complete the enumeration

````
[13275.086059] intel-sdw sdw-master-1: Slave attached, programming device number
[13275.098123] ata3: SATA link down (SStatus 4 SControl 300)
[13275.404047] atkbd serio0: Failed to enable keyboard on isa0060/serio0
[13275.583453] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[13275.583478] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[13276.083411] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[13276.083439] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[13277.108047] intel-sdw sdw-master-1: IO transfer timed out
[13277.108064] rt700 sdw:1:25d:700:0:0: [rt700_sdw_write] 3121 <= 0000
[13277.108321] intel-sdw sdw-master-1: Msg Ack not received
[13277.108327] intel-sdw sdw-master-1: Msg ignored for Slave 0
[13277.108338] intel-sdw sdw-master-1: No more devices to enumerate
[13277.108342] intel-sdw sdw-master-1: End of sdw_program_device_num ret 0
[13277.108424] intel-sdw sdw-master-1: Msg Ack not received
[13277.108428] intel-sdw sdw-master-1: Msg ignored for Slave 1
[13277.108436] rt700 sdw:1:25d:700:0:0: Unable to sync register 0x3501. -61
[13277.108521] intel-sdw sdw-master-1: Msg Ack not received
[13277.108525] intel-sdw sdw-master-1: Msg ignored for Slave 1
[13277.108529] rt700 sdw:1:25d:700:0:0: Unable to sync register 0x7520001a. -61
````

@bardliao can you look into this with @shumingfan?

**Comment by bardliao** [comment]
> @bardliao can you look into this with @shumingfan?

Sure

**Comment by bardliao** [comment]
@sinahuang Could you share ``stress-s3.sh`` ?

**Comment by YvonneYang2** [comment]
Script:
[stress-s3.sh.txt](https://github.com/thesofproject/linux/files/3770046/stress-s3.sh.txt)

**Comment by bardliao** [comment]
@sinahuang Could you retest it with https://github.com/thesofproject/linux/pull/1370? It works on my side.

**Comment by sinahuang** [comment]
verify  issue with soundware-latest ( commit :1c3de3d )+PR #1370  for 1000 times，original issue can't reproduce but some errors in dmesg shows as below:
occurs "panic" and  "failed:-110" at 163th time.

````
[ 4179.514863] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0f successful
[ 4179.514886] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0e successful
[ 4179.514890] sof-audio-pci 0000:00:1f.3: unstall/run core: core_mask = 1
[ 4179.514896] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 1 : core_mask 1
[ 4179.515414] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x80000000 successful
[ 4179.515425] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf010e0e successful
[ 4179.515574] sof-audio-pci 0000:00:1f.3: error: no reply expected, received 0x0
[ 4179.533425] atkbd serio0: Failed to deactivate keyboard on isa0060/serio0
[ 4179.605121] ata3: SATA link down (SStatus 4 SControl 300)
[ 4179.815471] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x6000010 timedout
[ 4179.815475] sof-audio-pci 0000:00:1f.3: error: cl_dsp_init: timeout HDA_DSP_SRAM_REG_ROM_STATUS read
[ 4179.815491] sof-audio-pci 0000:00:1f.3: unknown ROM status value 06000010
[ 4179.815507] sof-audio-pci 0000:00:1f.3: error: status = 0x00000000 panic = 0x00000000
[ 4179.815513] sof-audio-pci 0000:00:1f.3: unknown ROM status value 06000010
[ 4179.815523] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x1010f0f successful
[ 4179.815531] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x1000f0f successful
[ 4179.815537] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 0 : core_mask f
[ 4179.815544] sof-audio-pci 0000:00:1f.3: error: Error code=0xffffffff: FW status=0xffffffff
[ 4179.815546] sof-audio-pci 0000:00:1f.3: error: iteration 0 of Core En/ROM load failed: -110
[ 4179.816112] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0f successful
[ 4179.816138] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0e successful
[ 4179.816144] sof-audio-pci 0000:00:1f.3: unstall/run core: core_mask = 1
[ 4179.816151] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 1 : core_mask 1
[ 4179.817032] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x80000000 successful
[ 4179.817046] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf010e0e successful
[ 4179.817235] sof-audio-pci 0000:00:1f.3: error: no reply expected, received 0x0
[ 4179.878007] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x5000001 successful
[ 4179.888457] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x5 successful
[ 4179.889005] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x240000 successful
[ 4179.889019] sof-audio-pci 0000:00:1f.3: Firmware download successful, booting...
 
````
</details> 

occur "Link power down failed: -11" at 148th time.
 
````
ubuntu@sh-icl-rvp-hda-001:~/scripts/logs$ cat test_148.log
[ 3765.902035] intel-sdw sdw-master-1: intel_suspend_runtime start
[ 3765.902038] intel-sdw sdw-master-2: intel_suspend_runtime start
[ 3765.902116] intel-sdw sdw-master-1: intel_suspend_runtime done
[ 3765.902122] intel-sdw sdw-master-3: intel_suspend_runtime start
[ 3765.902199] intel-sdw sdw-master-3: intel_suspend_runtime done
[ 3765.902634] intel-sdw sdw-master-2: Link power down failed: -11
[ 3770.656917] PM: suspend entry (s2idle)
[ 3770.657689] Filesystems sync: 0.000 seconds
[ 3770.658144] Freezing user space processes ... (elapsed 0.002 seconds) done.
[ 3770.661094] OOM killer disabled.
[ 3770.661095] Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
[ 3770.662610] printk: Suspending console(s) (use no_console_suspend to debug)
[ 3770.663041]  SDW0-speakers: ASoC: pre trigger FE SDW0-speakers cmd 5
[ 3770.663048] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 0 dir 0 cmd 5
[ 3770.663050] sof-audio-pci 0000:00:1f.3: pcm: trigger suspend stream 0 dir 0 cmd 5
[ 3770.663056] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[ 3770.663221] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[ 3770.663694] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x140000 successful
[ 3770.663747] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[ 3770.663877] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[ 3770.663880]  SDW0-Playback: ASoC: trigger BE SDW0-Playback cmd 5
[ 3770.663882] intel-sdw sdw-master-0: intel_trigger: SDW0 Pin2: start
[ 3770.663882] intel-sdw sdw-master-0: intel_trigger: SDW0 Pin2: suspend
[ 3770.663883] sdw_disable_stream: SDW0 Pin2-Playback: start
[ 3770.664535] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00e0 <= 0003
[ 3770.664605] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00f0 <= 0003
[ 3770.664606] rt711 sdw:0:25d:711:0: rt711_clock_config complete, clk_freq=4800000
[ 3770.665270] sdw_disable_stream: SDW0 Pin2-Playback: done
 
````
[failed -11 log at 148th.zip](https://github.com/thesofproject/linux/files/3777001/failed.-11.log.at.148th.zip)
[panic and failed -110 log at 163th.zip](https://github.com/thesofproject/linux/files/3777002/panic.and.failed.-110.log.at.163th.zip)


**Test recipet**   
kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 1c3de3d +PR #1370
FW: https://github.com/thesofproject/sof/commits/master commit: 89ef0af 
tplg: Same with FW branch, sof-icl-rt711-rt1308-rt715-hdmi.tplg  rename  to  sof-icl-rt711-rt1308-rt715.tplg
platform: ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode

**Comment by sinahuang** [comment]
This issue still can be reproduced on TGL RVP with codec ACL711, ALC1308 in SDW mode.
Failed at 11th time.

Dmesg shows at 11th time as below:
```
[  974.769146] Restarting tasks ...
[  974.771409]  SDW0-headphone: ASoC: prepare FE SDW0-headphone
[  974.771419]  SDW0-Playback: ASoC: prepare BE SDW0-Playback
[  974.771429] intel-sdw sdw-master-0: intel_prepare: SDW0 Pin2: start
[  974.771471] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[  974.778255] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[  974.778259] sdw_prepare_stream: SDW0 Pin2-Playback start
[  974.778314] intel-sdw sdw-master-0: Msg Ack not received
[  974.778315] intel-sdw sdw-master-0: Msg ignored for Slave 1
[  974.778317] rt711 sdw:0:25d:711:0: DPN_PortCtrl register write failed for port 3
[  974.778320] intel-sdw sdw-master-0: Program transport params failed: -61
[  974.778322] intel-sdw sdw-master-0: Program params failed: -61
[  974.778324] Prepare for stream:SDW0 Pin2-Playback failed: -61
[  974.778326] sdw_prepare_stream: SDW0 Pin2-Playback: done
[  974.778328] intel-sdw sdw-master-0: intel_prepare: SDW0 Pin2: done
[  974.778331] intel-sdw sdw-master-0: ASoC: cpu DAI prepare error: -61
[  974.778333]  SDW0-Playback: ASoC: backend prepare failed -61
[  974.778450]  SDW0-headphone: ASoC: hw_free FE SDW0-headphone
[  974.778453] sof-audio-pci 0000:00:1f.3: pcm: free stream 0 dir 0
[  974.778471]  SDW0-Playback: ASoC: hw_free BE SDW0-Playback
[  974.778474] intel-sdw sdw-master-0: intel_hw_free: SDW0 Pin2: start
[  974.778476] sdw_deprepare_stream: SDW0 Pin2-Playback: start
[  974.778780] sdw_deprepare_stream: SDW0 Pin2-Playback: done
[  974.778795] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[  974.778898] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
 
```
Kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 661d261
FW: https://github.com/thesofproject/sof/commits/master commit: 78b022b
Tplg: Same with TGL FW branch commit: sof-tgl-rt711-rt1308-nohdmi.tplg rename to sof-tgl-rt711-rt1308.tplg


[s3_test_11_logger_all.log](https://github.com/thesofproject/linux/files/3778102/s3_test_11_logger_all.log)
[test_11.log](https://github.com/thesofproject/linux/files/3778110/test_11.log)

**Comment by bardliao** [comment]
@plbossart It seems that somehow SDW_SCP_INT1_PARITY is needed for IO transfer. I can get 100% IO transfer timed out if I set SDW_SCP_INT1_PARITY after 3000 ms. 

```
@@ -723,8 +724,8 @@ static int sdw_initialize_slave(struct sdw_slave *slave)
 	 * TODO: Read implementation defined interrupt mask
 	 * from Slave property
 	 */
-	val = SDW_SCP_INT1_IMPL_DEF | SDW_SCP_INT1_BUS_CLASH |
-					SDW_SCP_INT1_PARITY;
+	val = SDW_SCP_INT1_IMPL_DEF | SDW_SCP_INT1_BUS_CLASH/* |
+					SDW_SCP_INT1_PARITY*/;
 
 	/* Enable SCP interrupts */
 	ret = sdw_update(slave, SDW_SCP_INTMASK1, val, val);
@@ -1159,6 +1160,10 @@ int sdw_handle_slave_status(struct sdw_bus *bus,
 				dev_err(bus->dev,
 					"Slave %d initialization failed: %d\n",
 					i, ret);
+			msleep(3000);
+			sdw_update(slave, SDW_SCP_INTMASK1, SDW_SCP_INT1_PARITY, SDW_SCP_INT1_PARITY);
+
 
 			break;
```

**Comment by bardliao** [comment]
> This issue still can be reproduced on TGL RVP with codec ACL711, ALC1308 in SDW mode.
> Failed at 11th time.
> 
> Kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: [661d261](https://github.com/thesofproject/linux/commit/661d2612aabbd222c4b6af99df530ce3c37ff24a)
> FW: https://github.com/thesofproject/sof/commits/master commit: 78b022b
> Tplg: Same with TGL FW branch commit: sof-tgl-rt711-rt1308-nohdmi.tplg rename to sof-tgl-rt711-rt1308.tplg
> 
> [s3_test_11_logger_all.log](https://github.com/thesofproject/linux/files/3778102/s3_test_11_logger_all.log)
> [test_11.log](https://github.com/thesofproject/linux/files/3778110/test_11.log)


@sinahuang Could you test with PR #1370?

**Comment by plbossart** [comment]
> @plbossart It seems that somehow SDW_SCP_INT1_PARITY is needed for IO transfer. I can get 100% IO transfer timed out if I set SDW_SCP_INT1_PARITY after 3000 ms.
> 
> ```
> @@ -723,8 +724,8 @@ static int sdw_initialize_slave(struct sdw_slave *slave)
>  	 * TODO: Read implementation defined interrupt mask
>  	 * from Slave property
>  	 */
> -	val = SDW_SCP_INT1_IMPL_DEF | SDW_SCP_INT1_BUS_CLASH |
> -					SDW_SCP_INT1_PARITY;
> +	val = SDW_SCP_INT1_IMPL_DEF | SDW_SCP_INT1_BUS_CLASH/* |
> +					SDW_SCP_INT1_PARITY*/;
>  
>  	/* Enable SCP interrupts */
>  	ret = sdw_update(slave, SDW_SCP_INTMASK1, val, val);
> @@ -1159,6 +1160,10 @@ int sdw_handle_slave_status(struct sdw_bus *bus,
>  				dev_err(bus->dev,
>  					"Slave %d initialization failed: %d\n",
>  					i, ret);
> +			msleep(3000);
> +			sdw_update(slave, SDW_SCP_INTMASK1, SDW_SCP_INT1_PARITY, SDW_SCP_INT1_PARITY);
> +
>  
>  			break;
> ```

Wow, this is super weird. This could be a codec issue where somehow the interrupt is set prior to the resume, and the transfers remain bad until we explicitly enable the interrupt?

Can you try to see if this is tied with the fact that we retransmit commands automatically?
I don't have the data sheet so double check the value.

#define CDNS_MCP_CONFIG_MCMD_RETRY		0 // GENMASK(27, 24)

**Comment by sinahuang** [comment]
@bardliao I had verified the issue with PR #1370 on soundware-latest ( commit :1c3de3d ), please refer to the https://github.com/thesofproject/linux/issues/1344#issuecomment-546778446

**Comment by bardliao** [comment]
> @bardliao I had verified the issue with PR #1370 on soundware-latest ( commit :1c3de3d ), please refer to the [#1344 (comment)](https://github.com/thesofproject/linux/issues/1344#issuecomment-546778446)

@sinahuang Sorry, I meant on TGL RVP with codec ACL711, ALC1308 in SDW mode since you said it was failed on it. You didn't test it with #1370, right? We need to make sure the PR works on every platforms. Thanks.

**Comment by bardliao** [comment]
@plbossart set CDNS_MCP_CONFIG_MCMD_RETRY to 0 doesn't help. And sorry, setting SDW_SCP_INT1_PARITY after 3000ms doesn't 100% get IO transfer timed out. I will investigate it tomorrow.

**Comment by sinahuang** [comment]
This issue can't be reproduced with soundware-latest ( commit : 661d261)+PR #1370, test for 800 times on TGL RVP with codec ACL711, ALC1308 in SDW mode.



Kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 661d261+ PR #1370 
FW: https://github.com/thesofproject/sof/commits/master commit: 0e8cf66e
Tplg: Same with TGL FW branch commit: sof-tgl-rt711-rt1308-nohdmi.tplg rename to sof-tgl-rt711-rt1308.tplg

**Comment by YvonneYang2** [comment]
Same logs showed when run aplay0,0(RT711) stop/restart stress test. Failed at 322nd time.

<details>
 <summary>Dmesg</summary>

````
[ 1007.651957] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 1008.152213] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[ 1008.152240] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 1009.704233] intel-sdw sdw-master-0: IO transfer timed out
[ 1009.704508] intel-sdw sdw-master-0: Msg Ack not received
[ 1009.704517] intel-sdw sdw-master-0: Msg ignored for Slave 1
[ 1009.704528] rt711 sdw:0:25d:711:0: DPN_PortCtrl register write failed for port 3
[ 1009.704541] intel-sdw sdw-master-0: Program transport params failed: -61
[ 1009.704549] intel-sdw sdw-master-0: Program params failed: -61
[ 1009.704556] Prepare for stream:SDW0 Pin2-Playback failed: -61
[ 1009.704570] sdw_prepare_stream: SDW0 Pin2-Playback: done
[ 1009.704578] intel-sdw sdw-master-0: intel_prepare: SDW0 Pin2: done
[ 1009.704586] intel-sdw sdw-master-0: ASoC: cpu DAI prepare error: -61
[ 1009.704599]  SDW0-Playback: ASoC: backend prepare failed -61
[ 1009.704654] intel-sdw sdw-master-0: Slave attached, programming device number
[ 1009.704683]  Headphone: ASoC: pop wq checking: Playback status: active waiting: yes
[ 1009.704808]  Headphone: ASoC: hw_free FE Headphone
[ 1009.704824] sof-audio-pci 0000:00:1f.3: pcm: free stream 0 dir 0
[ 1009.704837] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[ 1009.704868] intel-sdw sdw-master-0: SDW Slave Addr: 20025d071100
[ 1009.704959] intel-sdw sdw-master-0: SDW Slave class_id 0, part_id 711, mfg_id 25d, unique_id 0, version 2
[ 1009.704973] intel-sdw sdw-master-0: in sdw_assign_device_num
[ 1009.704999] intel-sdw sdw-master-0: in sdw_assign_device_num
[ 1009.705016] intel-sdw sdw-master-0: Slave already registered, reusing dev_num:1
[ 1009.705030] intel-sdw sdw-master-0: in sdw_assign_device_num, writing dev_num 1
[ 1009.705176] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[ 1009.705202]  SDW0-Playback: ASoC: hw_free BE SDW0-Playback
[ 1009.705222] intel-sdw sdw-master-0: intel_hw_free: SDW0 Pin2: start
[ 1009.705245] sdw_deprepare_stream: SDW0 Pin2-Playback: start
[ 1009.705258] sdw_deprepare_stream: SDW0 Pin2-Playback: inconsistent state state 1
[ 1009.705269] sdw_deprepare_stream: SDW0 Pin2-Playback: done
[ 1009.705278] intel-sdw sdw-master-0: sdw_deprepare_stream: failed -22
[ 1009.705406]  SDW0-Playback: ASoC: close BE SDW0-Playback
[ 1009.705412] intel-sdw sdw-master-0: Msg Ack not received

````
</details>

Test recipe:
kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 9f1bd5b + PR1359
FW: https://github.com/thesofproject/sof/commits/master  commit: 2ca6a4a
tplg: https://github.com/bardliao/sof/commits/soundwire-tplg (commit: 0408153), sof-cml-rt711-rt1308-rt715.tplg
platform: CML SDW Laptop with ALC711+ ALC1308+ ALC715

stress script and logs:
[aplay_error_322_dmesg.log](https://github.com/thesofproject/linux/files/3787094/aplay_error_322_dmesg.log)
[aplay_error_322_logger_all.log](https://github.com/thesofproject/linux/files/3787095/aplay_error_322_logger_all.log)
[sof-test-playback.sh.txt](https://github.com/thesofproject/linux/files/3787096/sof-test-playback.sh.txt)

**Comment by bardliao** [comment]
@plbossart I think I got the reason of this issue. In the fail cases, codec is unattached after [wait_for_completion_timeout()](https://github.com/thesofproject/linux/blob/integration/soundwire-latest/sound/soc/codecs/rt711-sdw.c#L511) in rt711_dev_resume(). So we are actually accessing codec registers before codec is enumerated.

```
[ 1438.739780] rt711 sdw:0:25d:711:0: bard: rt711_dev_resume hw_init 1
...
[ 1438.739786] rt711 sdw:0:25d:711:0: bard: rt711_dev_resume sync start
...
[ 1438.739804] rt711 sdw:0:25d:711:0: sdw_modify_slave_status: initializing completion for Slave 1
[ 1438.739807] intel-sdw sdw-master-0: Slave attached, programming device number
[ 1438.740429] cdns_fill_msg_resp: 13 callbacks suppressed
[ 1438.740434] intel-sdw sdw-master-0: Msg Ack not received
[ 1438.740437] intel-sdw sdw-master-0: Msg ignored for Slave 1
[ 1438.740446] rt711 sdw:0:25d:711:0: Unable to sync register 0x3501. -61
[ 1438.740573] intel-sdw sdw-master-0: Msg Ack not received
[ 1438.740577] intel-sdw sdw-master-0: Msg ignored for Slave 1
[ 1438.740586] rt711 sdw:0:25d:711:0: Unable to sync register 0x752010. -61
[ 1438.740589] rt711 sdw:0:25d:711:0: bard: rt711_dev_resume return 0
...
[ 1438.740887] intel-sdw sdw-master-0: No more devices to enumerate
[ 1438.740890] intel-sdw sdw-master-0: End of sdw_program_device_num ret 0
[ 1438.740928] intel-sdw sdw-master-0: Slave status change
[ 1438.740932] intel-sdw sdw-master-0: sdw_handle_slave_status: start
[ 1438.740936] intel-sdw sdw-master-0: processing Slave 1 Status 0
[ 1438.740938] intel-sdw sdw-master-0: Updating Slave 1 status
[ 1438.740942] rt711 sdw:0:25d:711:0: bard: rt711_update_status status 0
[ 1438.740945] intel-sdw sdw-master-0: sdw_handle_slave_status: Updating Slave 1 status done
[ 1438.740947] intel-sdw sdw-master-0: sdw_handle_slave_status: end
...
[ 1438.741832] rt715 sdw:3:25d:715:0: sdw_modify_slave_status: initializing completion for Slave 1
```
[dmesg_fail.txt](https://github.com/thesofproject/linux/files/3787431/dmesg_fail.txt)

**Comment by bardliao** [comment]
@plbossart Do you think #1402 make sense?

**Comment by plbossart** [comment]
@bardliao no, sorry. the wait_for_enumeration() should only be successful if the Device1 is enumerated, but apparently the programming of the device1 fails.
Your traces also show that we don;t have the "signaling  completion' log, so I don't get how the wait_for_completion() could possibly succeed?

Generally if the programming of the device fails, it's due to another race where e.g. we reset the bus while there's traffic on-going.

I would suggest checking the logic for the init_completion, complete and wait_for_completion, that's where the issue is likely happening. I don't think doing a rt711-specific change is required.

**Comment by bardliao** [comment]
@ClarexZhou @sinahuang Could you retest it with #1407 ? Thanks.

**Comment by plbossart** [comment]
> @ClarexZhou @sinahuang Could you retest it with #1407 ? Thanks.

actually it'll be merged into integration/soundwire-latest by the time you reach your desk on Friday morning.

**Comment by YvonneYang2** [comment]
Checked it with MSI disabled on latest integration/soundwire-latest(commit:9f1bd5b) + #1403 + #1407 , original issue cannot be reproduced. But some failed info shows in dmesg.  

Run with command " aplay -Dhw:0,0 -r48000 -c2 -fs16_le /dev/zero & sudo ./stress-s3.sh 1000 1 1"
Platform: CML SDW Laptop with ALC711+ ALC1308+ ALC715 

Note: Since s2idle is blocked by #1387 , eg. cannot stay in suspended status(s2idle mode) over 1s. So suspend with deep mode for this test. 

<details>
 <summary>error dmesg before PM suspend exit</summary>

````
[  622.846660] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[  622.846688] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[  623.869622] rt715 sdw:3:25d:715:0: Enumeration not complete, timed out
[  623.869641] PM: dpm_run_callback(): rt715_dev_resume+0x0/0x90 [snd_soc_rt715] returns -110
[  623.869648] PM: Device sdw:3:25d:715:0 failed to resume: error -110
[  623.869670] sdw_rt711_rt1308_rt715 sdw_rt711_rt1308_rt715: ASoC: Scheduling resume work
[  623.869684] sdw_rt711_rt1308_rt715 sdw_rt711_rt1308_rt715: ASoC: starting resume work
[  623.870063] sdw_rt711_rt1308_rt715 sdw_rt711_rt1308_rt715: ASoC: resume work completed
[  623.870404] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 3501 <= 0003
[  623.870605] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 3501 <= 0000
[  623.870818] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 3603 <= 0010
[  623.871139] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 7321 83a1 <= b000
[  623.872923] acpi LNXPOWER:09: Turning OFF
[  623.873782] OOM killer enabled.
[  623.873782] Restarting tasks ... 
[  623.879129]  Headphone: ASoC: prepare FE Headphone
[  623.879134]  SDW0-Playback: ASoC: prepare BE SDW0-Playback
[  623.879141] intel-sdw sdw-master-0: intel_prepare: SDW0 Pin2: start
[  623.879158] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[  623.886071] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[  623.886074] sdw_prepare_stream: SDW0 Pin2-Playback start
[  623.888691] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00e0 <= 0001
[  623.888751] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00f0 <= 0001
[  623.888753] rt711 sdw:0:25d:711:0: rt711_clock_config complete, clk_freq=6000000
[  623.888856] sdw_prepare_stream: SDW0 Pin2-Playback: done
[  623.888861] intel-sdw sdw-master-0: intel_prepare: SDW0 Pin2: done
[  623.888888] sof-audio-pci 0000:00:1f.3: pcm: prepare stream 0 dir 0
[  623.888890] sof-audio-pci 0000:00:1f.3: pcm: hw params stream 0 dir 0
[  623.888898] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x40000 successful
[  623.888913] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x40000 successful
[  623.888915] sof-audio-pci 0000:00:1f.3: period_bytes:0x3fc0
[  623.888917] sof-audio-pci 0000:00:1f.3: periods:4
[  623.888927] sof-audio-pci 0000:00:1f.3: stream_tag 1
[  623.888933] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[  623.889132] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[  623.889134] sof-audio-pci 0000:00:1f.3: pcm: stream dir 0, posn mailbox offset is 790528
[  623.889149]  SDW0-Playback: ASoC: BE SDW0-Playback event 1 dir 0
[  623.889205]  SDW0-Playback: ASoC: trigger BE SDW0-Playback cmd 1
[  623.889207] intel-sdw sdw-master-0: intel_trigger: SDW0 Pin2: start
[  623.889209] sdw_enable_stream: SDW0 Pin2-Playback: start
[  623.896057] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00e0 <= 0001
[  623.896571] rt711 sdw:0:25d:711:0: [rt711_sdw_write] 00f0 <= 0001
[  623.896573] rt711 sdw:0:25d:711:0: rt711_clock_config complete, clk_freq=6000000
[  623.896869] sdw_enable_stream: SDW0 Pin2-Playback: done
[  623.896874] intel-sdw sdw-master-0: intel_trigger: SDW0 Pin2: done
[  623.896878]  Headphone: ASoC: post trigger FE Headphone cmd 1
[  623.896882] sof-audio-pci 0000:00:1f.3: pcm: trigger stream 0 dir 0 cmd 1
[  623.897299] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x14001e successful
[  623.897302] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60040000: GLB_STREAM_MSG: TRIG_START
[  623.898112] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60040000: GLB_STREAM_MSG: TRIG_START
[  623.904046] done.
[  623.943413] PM: suspend exit


````
</details>

<details>
 <summary>SCP Msg trf timed out</summary>

````
[  621.876721] rt711 sdw:0:25d:711:0: sdw_modify_slave_status: signaling completion for Slave 1
[  621.876722] intel-sdw sdw-master-0: Slave 1 status attached, was alert, ignoring
[  621.876722] intel-sdw sdw-master-0: Updating Slave 1 status
[  621.876723] intel-sdw sdw-master-0: sdw_handle_slave_status: Updating Slave 1 status done
[  621.876724] intel-sdw sdw-master-0: sdw_handle_slave_status: end
[  622.134325] rt711 sdw:0:25d:711:0: [rt711_sdw_read] b921 0000 => 80000000
[  622.134989] rt711 sdw:0:25d:711:0: [rt711_sdw_read] 7520 85a0 9c20 aca0 => 00000400
[  622.135613] rt711 sdw:0:25d:711:0: [rt711_sdw_read] 7520 85a0 9c20 aca0 => 00000090
[  622.135617] rt711 sdw:0:25d:711:0: in rt711_jack_detect_handler, jack_type=0x3
[  622.135620] rt711 sdw:0:25d:711:0: in rt711_jack_detect_handler, btn_type=0x0
[  622.141322] intel-sdw sdw-master-1: SCP Msg trf timed out
[  622.141509] intel-sdw sdw-master-1: Program SCP Ack not received
[  622.141513] intel-sdw sdw-master-1: Program SCP Ack not received
[  622.141517] cdns_program_scp_addr: 582 callbacks suppressed
[  622.141521] intel-sdw sdw-master-1: SCP_addrpage ignored for Slave 1
[  622.141621] intel-sdw sdw-master-1: Program SCP Ack not received
[  622.141623] intel-sdw sdw-master-1: Program SCP Ack not received
[  622.141626] intel-sdw sdw-master-1: SCP_addrpage ignored for Slave 1
[  622.141724] intel-sdw sdw-master-1: Program SCP Ack not received
[  622.141726] intel-sdw sdw-master-1: Program SCP Ack not received
[  622.141728] intel-sdw sdw-master-1: SCP_addrpage ignored for Slave 1
[  622.141831] intel-sdw sdw-master-1: Program SCP Ack not received
[  622.141833] intel-sdw sdw-master-1: Program SCP Ack not received


````
</details>

Logs:
[s3_test_32_logger_error.zip](https://github.com/thesofproject/linux/files/3796464/s3_test_32_logger_error.zip)

**Comment by YvonneYang2** [comment]
Test it with MSI mode, same failed dmesg info as above can still be reproduced.  

Platform: CML SDW Laptop with ALC711+ ALC1308+ ALC715 

Logs:
[s3_test_28_logger_all.log](https://github.com/thesofproject/linux/files/3796621/s3_test_28_logger_all.log)
[s3_test_28_logger_error.log](https://github.com/thesofproject/linux/files/3796622/s3_test_28_logger_error.log)
[test_28.log](https://github.com/thesofproject/linux/files/3796623/test_28.log)

**Comment by YvonneYang2** [comment]
Issue can still be reproduced when run aplay0,2 (RT1308) stop/restart stress test. Failed at 14th time.
Run with command " ./sof-test-playback.sh"

Script:
[sof-test-playback.sh.txt](https://github.com/thesofproject/linux/files/3822851/sof-test-playback.sh.txt)

<details>
 <summary>dmesg</summary>

````

[  210.177914] intel-sdw sdw-master-1: sdw_handle_slave_status: start
[  210.674899] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[  210.674927] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[  211.174886] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[  211.174920] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[  212.199492] intel-sdw sdw-master-1: IO transfer timed out
[  212.199717] intel-sdw sdw-master-1: Msg Ack not received
[  212.199725] intel-sdw sdw-master-1: Msg ignored for Slave 1
[  212.199734] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[  212.199744] intel-sdw sdw-master-1: Program transport params failed: -61
[  212.199750] intel-sdw sdw-master-1: Program params failed: -61
[  212.199755] Prepare for stream:SDW1 Pin2-Playback failed: -61
[  212.199767] sdw_prepare_stream: SDW1 Pin2-Playback: done
[  212.199774] intel-sdw sdw-master-1: intel_prepare: SDW1 Pin2: done
[  212.199781] intel-sdw sdw-master-1: ASoC: cpu DAI prepare error: -61
[  212.199788]  SDW1-Playback: ASoC: backend prepare failed -61
[  212.199834] intel-sdw sdw-master-1: Slave attached, programming device number
[  212.199963]  SDW1-speakers: ASoC: hw_free FE SDW1-speakers
[  212.199975] sof-audio-pci 0000:00:1f.3: pcm: free stream 2 dir 0
[  212.199987] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[  212.200038] intel-sdw sdw-master-1: SDW Slave Addr: 20025d130800
[  212.200046] intel-sdw sdw-master-1: SDW Slave class_id 0, part_id 1308,
````
</details>

Test recipe:
kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: b12acfe 
FW: https://github.com/thesofproject/sof/commits/master  commit: 5211799
tplg: same path as FW branch, sof-cml-rt711-rt1308-mono-rt715.tplg
platform: CML SDW Laptop with ALC711+ ALC1308+ ALC715 

[aplay_error_14_dmesg.log](https://github.com/thesofproject/linux/files/3822855/aplay_error_14_dmesg.log)
[aplay_error_14_logger_all.log](https://github.com/thesofproject/linux/files/3822856/aplay_error_14_logger_all.log)

**Comment by bardliao** [comment]
``[  196.135457] rt1308 sdw:1:25d:1308:0: Initialization not complete, timed out``
I think it is the problem. I will look into it.

**Comment by YvonneYang2** [comment]
@bardliao 
Same dmesg log shows when run aplay0,0 (RT711) stop/restart stress test. Failed at 199th time.

Original issue reproduced steps cannot be tested since suspend with s2idle mode is blocked by https://github.com/thesofproject/linux/issues/1387 , 
```
[ 1399.186995]  Headphone: ASoC: found 1 new BE paths
[ 1399.186996]  SDW0-Playback: ASoC: open playback BE SDW0-Playback
[ 1401.191632] rt711 sdw:0:25d:711:0: Initialization not complete, timed out
[ 1401.191654] intel-sdw sdw-master-0: intel_startup: SDW0 Pin2: start
[ 1401.191666] intel-sdw sdw-master-0: intel_startup: SDW0 Pin2: done
[ 1401.191678]  Headphone: ASoC: open FE Headphone
[ 1401.191691] sof-audio-pci 0000:00:1f.3: pcm: open stream 0 dir
```

[aplay_error_199_dmesg.log](https://github.com/thesofproject/linux/files/3823740/aplay_error_199_dmesg.log)
[aplay_error_199_logger_all.log](https://github.com/thesofproject/linux/files/3823741/aplay_error_199_logger_all.log)

**Comment by bardliao** [comment]
@plbossart Somehow there is no intel_resume_runtime around the "Initialization not complete, timed out" (rt1308_dev_resume)
The following trace is from aplay_error_14_dmesg.log
```
[  189.773575] intel-sdw sdw-master-1: intel_resume_runtime done
...
[  196.135457] rt1308 sdw:1:25d:1308:0: Initialization not complete, timed out
...
[  210.177000] intel-sdw sdw-master-1: intel_resume_runtime done
```

**Comment by bardliao** [comment]
@plbossart I think the reason why there is no intel_resume_runtime around rt1308_dev_resume is that 
 we call [pm_runtime_mark_last_busy()](https://github.com/thesofproject/linux/blob/integration/soundwire-latest/drivers/soundwire/intel.c#L1024) in intel_shutdown(), but not in codec's shutdown dai ops. I tried to do the same on rt1308-sdw.c but it doesn't solve the issue. On second thought, I think we can't assume that master is always suspended when a slave is suspended. For example, master will be active if there are two slaves on the same sdw bus, one is suspended and the other is active. So probably we need to find a way to know if we have to wait for initialization_complete in codec resume or not.

**Comment by plbossart** [comment]
tthanks @bardlio, good analysis

so the sequence is

(1) resume, playback, stop
(2) intel_shutdown() gets called, we call put_autosuspend()
(3) timeout in rt1308_dev_resume()
(4)playback restarts, we call intel_startup() and call pm_runtime_get()
  
I guess what happens is that the DAIs are released, so the codec suspends, but the master does not suspend so as a result we will not force the codec to be tagged as unattached and we don't call complete(), ever.

what we'd need to have is something like is_bus_suspended() to figure out if the context was lost or not.

For now, can we try the stress test with a longer interval between each playback (make that 10s) to make sure the master will suspend?

**Comment by plbossart** [comment]
@bardliao @YvonneYang2 PR #1459 should solve this bug, please double-check.

**Comment by bardliao** [comment]
> @bardliao @YvonneYang2 PR #1459 should solve this bug, please double-check.

@plbossart I tested ./sof-test-playback.sh and it passed for 500 times

**Comment by plbossart** [comment]
@sinahuang   @mengdonglin integration/soundwire-latest now has all the known fixes for suspend/resume please retry (along with the latest firmware)

**Comment by YvonneYang2** [comment]
Error not seen in 500 times aplay stress test. 
Original steps suspend/resume stress test in MSI mode is still on going.

Test recipe:
kernel: https://github.com/thesofproject/linux/tree/integration/soundwire-latest commit: 2933f73 
FW: https://github.com/thesofproject/sof/commits/master  commit: d1dcdfe
tplg: same path as FW branch, sof-cml-rt711-rt1308-mono-rt715.tplg
platform: CML SDW Laptop with ALC711+ ALC1308+ ALC715

**Comment by YvonneYang2** [comment]
Original issue cannot be reproduced in 300 times suspend/resume with playback stress test. But occurred "IO transfer timed out" at 125th/126th/138/139th time.
Besides, some error info when do s2idle suspend with aplay is same as https://github.com/thesofproject/linux/issues/1387#issuecomment-552753463, but aplay function will not be effected.

**IO transfer timed out**
```
[ 2523.856159] PM: suspend exit
[ 2524.301138] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[ 2524.301793] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 2524.801069] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[ 2524.801690] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 2525.818178] intel-sdw sdw-master-1: IO transfer timed out
[ 2525.818195] intel-sdw sdw-master-1: SDW Slave Addr: 20025d130800
[ 2525.818200] intel-sdw sdw-master-1: SDW Slave class_id 0, part_id 1308, mfg_id 25d, unique_id 0, version 2
[ 2525.818203] intel-sdw sdw-master-1: in sdw_assign_device_num
[ 2525.818206] intel-sdw sdw-master-1: in sdw_assign_device_num
[ 2525.818209] intel-sdw sdw-master-1: Slave already registered, reusing dev_num:1
[ 2525.818213] intel-sdw sdw-master-1: in sdw_assign_device_num, writing dev_num 1
```

**error info**
```
[ 2844.104927] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0e0e successful
[ 2844.104931] sof-audio-pci 0000:00:1f.3: unstall/run core: core_mask = 1
[ 2844.104969] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 1 : core_mask 1
[ 2844.605212] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x0 timedout
[ 2844.605218] sof-audio-pci 0000:00:1f.3: error: cl_dsp_init: timeout for HIPCIE done
[ 2844.605225] sof-audio-pci 0000:00:1f.3: unknown ROM status value ffffffff
[ 2844.605231] sof-audio-pci 0000:00:1f.3: error: status = 0xffffffff panic = 0xffffffff
[ 2844.605236] sof-audio-pci 0000:00:1f.3: unknown ROM status value ffffffff
[ 2844.605245] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0f successful
[ 2844.605252] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f successful
[ 2844.605257] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 0 : core_mask f
[ 2844.605263] sof-audio-pci 0000:00:1f.3: error: Error code=0xffffffff: FW status=0xffffffff
[ 2844.605267] sof-audio-pci 0000:00:1f.3: error: iteration 0 of Core En/ROM load failed: -110
[ 2844.605273] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0f successful
[ 2844.605308] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0xf0f0f0e successful
[ 2844.605312] sof-audio-pci 0000:00:1f.3: unstall/run core: core_mask = 1
[ 2844.605318] sof-audio-pci 0000:00:1f.3: DSP core(s) enabled? 1 : core_mask 1
[ 2844.605918] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x80000000 successful
```

Logs:
[s3_test_125_logger_all.log](https://github.com/thesofproject/linux/files/3839057/s3_test_125_logger_all.log)
[test_125.log](https://github.com/thesofproject/linux/files/3839058/test_125.log)
[s3_test_126_logger_all.log](https://github.com/thesofproject/linux/files/3839062/s3_test_126_logger_all.log)
[test_126.log](https://github.com/thesofproject/linux/files/3839063/test_126.log)

Script:
[stress-s3.sh.txt](https://github.com/thesofproject/linux/files/3839603/stress-s3.sh.txt)
Run with command "aplay -Dhw:0,0 -r48000 -c2 -fs16_le /dev/zero & sudo ./stress-s3.sh 300 1 1 "

**Comment by bardliao** [comment]
I did the same test, but didn't see IO transfer timed out on my side. I tested it on https://github.com/thesofproject/linux/commit/fb40b85ac3d3435d3a05b8a59bf0409e80ccbfa0

**Comment by YvonneYang2** [comment]
Run 300 times aplay+suspend/resume stress test with MSI IRQ mode, IO transfer timed out still can be reproduced on my side. Tested it on fb40b85

**Comment by YvonneYang2** [comment]
Since original issue cannot be reproduced, close it and track IO transfer timed out issue in https://github.com/thesofproject/linux/issues/1482
